pkgs/by-name/sq/sqlline/package.nix

```
@@ -0,0 +1,55 @@
+{
+  lib,
+  maven,
+  fetchFromGitHub,
+  makeWrapper,
+  jre,
+  docbook_xml_dtd_42,
+}:
+
+maven.buildMavenPackage rec {
+  pname = "sqlline";
+  version = "1.12.0";
+
+  src = fetchFromGitHub {
+    owner = "julianhyde";
+    repo = "sqlline";
+    rev = "sqlline-${version}";
+    hash = "sha256-rUlGtMgTfhciQVif0KaUcuY28wh+PrHsKen8qODom24=";
+  };
+
+  mvnHash = "sha256-9qDzc6TRn9Yv3/nTATZgP6J+PTZEZCN1et/3GrRb7X4=";
+
+  nativeBuildInputs = [
+    makeWrapper
+    docbook_xml_dtd_42
+  ];
+
+  mvnParameters = "-DskipTests";
+
+  # Patch the DOCTYPE declaration in manual.xml
+  postPatch = ''
+    sed -i 's|https://docbook.org/xml/4.2/docbookx.dtd|${docbook_xml_dtd_42}/xml/dtd/docbook/docbookx.dtd|' src/docbkx/manual.xml
+    sed -i 's|PUBLIC "-//OASIS//DTD DocBook XML V4.1.2//EN"|PUBLIC "-//OASIS//DTD DocBook XML V4.2//EN"|' src/docbkx/manual.xml
```

Don't `sed` when you can `substituteInPlace` (mostly for better error reporting)
```suggestion
    substituteInPlace src/docbkx/manual.xml \
      --replace-fail "https://docbook.org/xml/4.2/docbookx.dtd" "${docbook_xml_dtd_42}/xml/dtd/docbook/docbookx.dtd" \
      --replace-fail 'PUBLIC "-//OASIS//DTD DocBook XML V4.1.2//EN"' 'PUBLIC "-//OASIS//DTD DocBook XML V4.2//EN"'
```